ProjFS.Mac/PrjFSKext/VnodeCache.cpp

```diff
@@ -383,10 +454,12 @@ KEXT_STATIC bool TryFindVnodeIndex_Locked(
         if (vnodeIndex == vnodeHashIndex)
         {
             // Looped through the entire cache and didn't find an empty slot or the vnode
+            AtomicFetchAddCacheHealthStat(VnodeCacheHealthStat_TotalLookupCollisions, totalSteps);
```

Comment:
> In what case is this not just
> AtomicFetchAddCacheHealthStat(VnodeCacheHealthStat_TotalLookupCollisions, s_entriesCapacity);

`s_entriesCapacity` is the total capacity of the vnode cache.  `totalSteps` is the number of steps we had to take after our hash function produces a collision.  

Usually these will be different because we won't have to linearly probe the entire cache to find the entry we're looking for (or an empty slot depending on if we're looking up an entry or inserting).

> Could we drop totalSteps and the additional preincrement in the loop?

We could, but it would mean that we'd potentially introduce more atomic operations.  By using `totalSteps` we ensure that we'll only atomically adjust `VnodeCacheHealthStat_TotalLookupCollisions` once per call to `TryFindVnodeIndex_Locked`.

(I can add a comment to totalSteps explaining the motivation if that would be helpful)

Reply:
Ah, I see where I was misreading this -- I completely missed the add on ln461.I was confused why we needed to keep track of the variable when, _at this exit branch_, it should always report `s_entriesCapacity` since we probed the entire table and didn't find room. No change necessary, reviewer problem :)

Reply:
Oh sorry I missed that this comment was for this case:

`// Looped through the entire cache and didn't find an empty slot or the vnode`

In that case I would expect `totalSteps` to equal `s_entriesCapacity` (I went with `totalSteps` because at least for me it was easier to follow the code when we used `totalSteps` here and on ln461